auto_nag/scripts/inactive_ni_pending.py

```diff
@@ -0,0 +1,184 @@
+                else:
+                    requestee_bugs[flag["requestee"]].append(bugid)
+
+        user_activity = UserActivity()
+        inactive_users = user_activity.check_users(requestee_bugs.keys())
+        selected_bugs = []
+        for requestee, bugids in requestee_bugs.items():
+            if requestee in inactive_users:
+                selected_bugs.extend(bugids)
+
+        for bugid in set(bugs.keys()) - set(selected_bugs):
+            del bugs[bugid]
+
+        for bug in bugs.values():
+            bug["inactive_ni"] = [
+                {
+                    "id": flag["id"],
+                    "requestee": flag["requestee"],
+                    "requestee_status": user_activity.get_string_status(
+                        inactive_users[flag["requestee"]]
+                    ),
+                }
+                for flag in bug["needinfo_flags"]
+                if flag["requestee"] in inactive_users
+            ]
+            bug["inactive_ni_count"] = len(bug["inactive_ni"])
+            self.add_action(bug)
+
+    def is_recent_bug(self, bug):
+        return lmdutils.get_date_ymd(bug["creation_time"]) >= self.__recent_bug_limit
+
+    def add_action(self, bug):
+        users_num = len(set([flag["requestee"] for flag in bug["inactive_ni"]]))
+
+        if (
+            (
+                bug["priority"] in HIGH_PRIORITY
+                or bug["severity"] in HIGH_SEVERITY
+                or self.is_recent_bug(bug)
+            )
+            and not utils.is_no_assignee(bug["triage_owner"])
+            and not any(
+                [
+                    flag["requestee"] == bug["triage_owner"]
+                    for flag in bug["inactive_ni"]
+                ]
+            )
+        ):
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "requestee": bug["triage_owner"],
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Redirect needinfo pending on inactive {utils.plural('user', users_num)} to the triage owner.",
+                },
+            }
+        else:
+            autofix = {
+                "flags": [
+                    {
+                        "id": flag["id"],
+                        "status": "X",
+                    }
+                    for flag in bug["inactive_ni"]
+                ],
+                "comment": {
+                    "body": f"Clear needinfo pending on inactive {utils.plural('user', users_num)}.",
+                },
+            }
+
+        self.add_prioritized_action(bug, bug["triage_owner"], autofix=autofix)
+
+    def get_bug_sort_key(self, bug):
+        return (
+            bug["priority"] not in HIGH_PRIORITY,
+            bug["severity"] not in HIGH_SEVERITY,
+            not self.is_recent_bug(bug),
```

Comment:
We could simply sort by creation date

Reply:
Maybe could even be part of the default `get_sort_by_bug_importance_key` since it makes sense by default for tools to consider more recent as more important

Reply:
> We could simply sort by creation date

If we are going to use the raw value of the creation date, it should be the last factor in the sorting key. Otherwise, bugs will be sorted mainly by the creation date. Later factors will only be considered when two bugs have the same exact creation date (it could be mitigated by using the month instead of the full date).

> Maybe could even be part of the default `get_sort_by_bug_importance_key` since it makes sense by default for tools to consider more recent as more important

It will make `get_sort_by_bug_importance_key()` more complicated, especially when the threshold is need to be configured (it could be solved by using the raw value without a threshold as you suggested). Also, this will makes`get_sort_by_bug_importance_key()` to be always requiring the `creation_time` field. Finally, this will have little impact, it will be used only when a bug has equal severity and priority values.